pkg/memtier/proc.go

```diff
@@ -432,7 +439,7 @@ func (f *ProcPagemapFile) ForEachPage(addressRanges []AddrRange, pageAttributes
 		readBuf := make([]byte, 16*(1+f.readahead))
 		readData := readBuf[0:0] // valid data in readBuf
 		for pageIndex := uint64(0); pageIndex < addressRange.length; pageIndex++ {
-			if len(readData) == 0 && f.read(&readData, &readBuf, &pagemapOffset, &addressRange, pageIndex) == 0 {
+			if f.tryToReadData(&readData, &readBuf, &pagemapOffset, &addressRange, pageIndex) {
```

Comment:
Let's not add here a function and a call just to remove one single "&&" from the if on line 442.